ledger/db/src/mint_config_store.rs

```diff
@@ -139,8 +141,38 @@ impl MintConfigStore {
 
         // Update active mint configurations.
         for set_mint_config_tx in set_mint_config_txs {
-            self.set_active_mint_configs(set_mint_config_tx, db_transaction)?;
+            // All mint configurations must have the same token id.
+            if set_mint_config_tx
+                .prefix
+                .configs
+                .iter()
+                .any(|mint_config| mint_config.token_id != set_mint_config_tx.prefix.token_id)
+            {
+                return Err(Error::InvalidMintConfig(
+                    "All mint configurations must have the same token id".to_string(),
+                ));
+            }
+
+            // MintConfigs -> ActiveMintConfigs
+            let active_mint_configs = ActiveMintConfigs::from(set_mint_config_tx);
+
+            // Store in database
+            db_transaction.put(
+                self.block_index_by_set_mint_config_tx_nonce,
+                &set_mint_config_tx.prefix.nonce,
+                &block_index_bytes,
+                WriteFlags::NO_OVERWRITE, /* this ensures we do not overwrite a nonce that was
+                                           * already used */
```

Comment:
Nit: Switch to line comment(s) above the value.
```suggestion
                //  ensures we do not overwrite a nonce that was already used
                WriteFlags::NO_OVERWRITE,
```